homeassistant/components/energenie/config_flow.py

```diff
@@ -0,0 +1,53 @@
+"""Config flow for Energenie integration."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_NAME
+from homeassistant.exceptions import HomeAssistantError
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_NAME, default="Energenie Socket"): str,
+        vol.Required("socket_number", default=1): int,
+    }
+)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for energenie."""
+
+    async def in_range(self, socket_number):
+        """Verify that input is between the range of 1 and 4."""
+        if 1 <= socket_number <= 4:
+            return True
+        raise InvalidRange
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+        if user_input is not None:
+            try:
+                await self.in_range(user_input["socket_number"])
+                await self.async_set_unique_id(f'energenie-socket-{user_input["socket_number"]}')
+                self._abort_if_unique_id_configured()
+                return self.async_create_entry(
```

Comment:
We want to check that we can connect to the device before creating the entry.

Reply:
There isn't a way to get confirmation from the device if you've successfully paired or not. It just sends a broadcast and if it's successful it turns on or turns off

Reply:
What happens if we set up the integration without the pi-mote controller board connected to the RPi or if we enter the wrong socket number?

Reply:
If the pi-mote isn't detected an exception is triggered: `gpiozero.exc.BadPinFactory: Unable to load any default pin factory!`.

The same case is when a wrong socket number is entered: `gpiozero.exc.EnergenieBadSocket: socket number must be between 1 and 4`

Reply:
So we could create the device instance and catch `gpiozero.exc.BadPinFactory`?

Reply:
Yep, this will allow us to confirm that the device has both the capabilities to run the board along if it can communicate. But what we wouldn't know is any of the sockets will receive the payload the board sends

Reply:
Ok, but it's better than nothing. :+1: